packages/at_client/lib/src/encryption_service/abstract_atkey_encryption.dart

```diff
@@ -159,6 +133,100 @@ abstract class AbstractAtKeyEncryption implements AtKeyEncryption {
     }
   }
 
```

Comment:
New method comment below describes the desired new logic for if / when a new shared symmetric key needs to be created